ci-operator/step-registry/optional-operators/cvp-common/scorecard/optional-operators-cvp-common-scorecard-commands.sh

```diff
@@ -5,47 +5,72 @@ set -o errexit
-        local attempt_num=1
-        until [[ $(jq . "$2") ]]
-        
-        do
-            "$1"
-            if (( attempt_num==retries_max ))
-            then
-                    echo "Retry attempt number: $attempt_num of $retries_max of scorecard tests failed. No more attempts. "
-                    return 1
-            else
-                    echo "Retry attempt number: $attempt_num of $retries_max failed. Retrying."
-                    sleep 5
-                    ((attempt_num++))
-            fi
-        done
+  local retries_max=3
+  local attempt_num=1
+  until [ -s "$2" ]
+
+  do
+    if (( attempt_num>retries_max )); then
+      echo "All $retries_max attempts of scorecard tests failed. No more attempts. "
+      return 1
+    else
+      echo "Attempt $attempt_num of $retries_max."
+      sleep 5
+      ((attempt_num++))
+    fi
+      "$1" "$2"
+  done
 }
 
+# Runs the basic scorecard tests using the prepared scorecard config
+# Takes 1 argument as the output file for the scorecard command
 basic_tests() {
-operator-sdk scorecard --config "${SCORECARD_CONFIG}" \
+  OUTPUT_FILE=$1
+  operator-sdk scorecard --config "${SCORECARD_CONFIG}" \
                        --namespace "${NAMESPACE}" \
                        --kubeconfig "${KUBECONFIG}" \
                        --verbose \
                        --output json \
-                       "${OPERATOR_DIR}" > "${ARTIFACT_DIR}"/scorecard-output-basic.json || true
+                       "${OPERATOR_DIR}" > "${OUTPUT_FILE}" || true
 }
 
+# Runs the custom scorecard using the config in the operator bundle image
+# Takes 1 argument as the output file for the scorecard command
+# If the CUSTOM_SCORECARD_TESTCASE is set, runs in single test case mode:
+#    Outputs the results in xunit format and stores them in the ARTIFACT_DIR
+#    Uses the selector option to execute just a single test from the config
 custom_tests() {
-operator-sdk scorecard \
-    --namespace="${NAMESPACE}" \
-    --kubeconfig "${KUBECONFIG}" \
-    --verbose \
-    --output json \
-    --wait-time 3000s \
-    --service-account "${SCORECARD_SERVICE_ACCOUNT}" \
-    "${OPERATOR_DIR}" > "${ARTIFACT_DIR}"/scorecard-output-custom.json || true
+  OUTPUT_FILE=$1
+  if [ -n "${CUSTOM_SCORECARD_TESTCASE}" ]; then
+    operator-sdk scorecard \
+        --namespace="${NAMESPACE}" \
+        --kubeconfig "${KUBECONFIG}" \
+        --verbose \
+        --output xunit \
+        --wait-time 3000s \
+        --test-output "${ARTIFACT_DIR}" \
+        --service-account "${SCORECARD_SERVICE_ACCOUNT}" \
+        --selector=test="${CUSTOM_SCORECARD_TESTCASE}" \
+        "${OPERATOR_DIR}" > "${OUTPUT_FILE}" || true
+  else
+    operator-sdk scorecard \
+        --namespace="${NAMESPACE}" \
+        --kubeconfig "${KUBECONFIG}" \
+        --verbose \
+        --output json \
+        --wait-time 3000s \
+        --service-account "${SCORECARD_SERVICE_ACCOUNT}" \
+        "${OPERATOR_DIR}" > "${OUTPUT_FILE}" || true
+  fi
```

Comment:
There is a lot of copy&paste here.

We can simplify that e.g.
```
EXTRA_PARAMETER=""
if [ # some condition ]; then
    EXTRA_PARAMETER="--arg1=value2"
fi
    operator-sdk scorecard \
        --namespace="${NAMESPACE}" \
        --kubeconfig "${KUBECONFIG}" \
        --verbose \
        --wait-time 3000s \
        --service-account "${SCORECARD_SERVICE_ACCOUNT}" \
        "$EXTRA_PARAMETER" \
        "${OPERATOR_DIR}" > "${OUTPUT_FILE}" || true
```

`-output xunit` must not be tight to `CUSTOM_SCORECARD_TESTCASE`. The type of output json/xunit have to controlled independently to `CUSTOM_SCORECARD_TESTCASE`. It allows better re-usability and flexibility.

Reply:
We may need to make this more reusable in a follow-on story. 

At this point only MSP will populate `CUSTOM_SCORECARD_TESTCASE` and therefore require xunit output. 

Unless the `TEST_MODE` variable that's used in the job config can be available to this script.

Reply:
@dirgim Yes, that'll work

Reply:
We should be able to manually trigger any combination.
e.g.
* test_mode={cvp,msp}
* output format ={json,xml}
* custom_scorecard test={"", "something"}

It will allow cvp developers to tests things more efficiently.
* you can use cvp mode (standard rh bundle) to test xunit output
* you can use cvp mode (different bundle e.g. sample one in our lower env) to test `CUSTOM_SCORECARD_TESTCASE`

But in real prod deployment; we will never pass combination mode=cvp && output type=xml.
Or we would be able to easily change that without touching files here.

Reply:
We currently can't test all the combinations if we launch with `cvp-trigger`, which currently accepts only `--custom-scorecard-testname` flag.

Reply:
That is a huge limitation which have to be fixed.

Reply:
@nmars I would even consider it to be a bug. Please file one.